gulpfile.js

```diff
@@ -82,6 +83,68 @@ function createReloadServer() {
   });
 }
 
+function writeServiceWorkerFile() {
+  let path = require('path');
+  let rootDir = 'dist';
+
+  let config = {
+    cacheId: 'polymerproject',
+    staticFileGlobs: [
+      `${rootDir}/1.0/index.html`,
+      `${rootDir}/1.0/shell.html`,
+      `${rootDir}/1.0/about.html`,
+      `${rootDir}/elements/**`,
+      `${rootDir}/js/*.js`,
+      `${rootDir}/css/*.css`,
+      `${rootDir}/bower_components/**/webcomponents-lite.min.js`,
+    ],
+    dynamicUrlToDependencies: {
+      '/1.0/shell.html': [
+        path.join(rootDir, 'templates', 'base-devguide.html'),
```

Comment:
As far as I understand it, those templates are used to generate the various doc pages that we cache using the runtime caching strategy. Theoretically, when the service worker is in charge, we load the shell, then replace the `innerHTML` of the `pw-shell` with the `innerHTML` from the doc page's `pw-shell`. 

When those files change, the _shell_ hasn't changed, just stuff that gets injected inside the `pw-shell`. AFAICT.

An yes, we probably could have designed this better for Service Worker :cry: 

On the plus side, the current site kind-of works with JavaScript turned off. So that's something.